save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DownloadFilesController.kt

```diff
@@ -182,10 +182,16 @@ class DownloadFilesController(
     @ApiResponse(responseCode = "404", description = "File is not found.")
     @PostMapping(path = ["/internal/files/download-save-agent"], produces = [MediaType.APPLICATION_OCTET_STREAM_VALUE])
     // FIXME: backend should set version of save-agent here for agent
-    fun downloadSaveAgent(): Mono<out Resource> =
-            Mono.just(ClassPathResource("save-agent.kexe"))
-                .filter { it.exists() }
-                .switchIfEmptyToNotFound()
+    fun downloadSaveAgent(): Mono<out Resource> {
+        val executable = "save-agent.kexe"
+
+        return Mono.just(ClassPathResource(executable))
+            .filter { it.exists() }
```

Comment:
Maybe so (and so on)? In order to keep the endpoint to be fully "chained"
```suggestion
    fun downloadSaveAgent(): Mono<out Resource> =  Mono.just("save-agent.kexe")

        .map { ClassPathResource(it) }
        .filter { it.exists() }
```

Reply:
+1
even more -- it can be extracted to a common private method and reused in two methods

Reply:
Addressed with d4bfadba7521d96c62a41600325762e1f23920c9